test/declaration_test.dart

```diff
@@ -517,6 +525,74 @@ div {
   expect(prettyPrint(styleSheet), expected);
 }
 
+void testSupports() {
```

Comment:
Great tests, maybe a negative test for a conjunction not having proper parens nesting (catching the error)?